hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/metadata/FlinkHoodieBackedTableMetadataWriter.java

```diff
@@ -133,7 +133,7 @@ protected void commitInternal(String instantTime, Map<String, HoodieData<HoodieR
       metadataMetaClient = HoodieTableMetaClient.reload(metadataMetaClient);
     }
 
-    compactIfNecessary(writeClient);
+    metadataWriteHandler.compactIfNecessary(writeClient, "");
```

Comment:
No changes are made in Flink MW

Reply:
We are assuming Flink table version is always above 6?

Reply:
Addressed